pysph/tools/ipy_viewer.py

```diff
@@ -634,29 +732,48 @@ def _legend_handler(self, change):
                                 0.82
                             ]
                     )
-                    maxm = np.max(c)
-                    minm = np.min(c)
-                    if (minm == maxm == 0):
-                        boundaries = np.linspace(0, 1, 100)
+
+                    arr = self._normalized_for_colormaps(c)
+                    minm = arr[0]
+                    maxm = arr[1]
+                    absmaxm = arr[2]
+
+                    # Do not change the order of the if-else #
+                    # block below. It is important that the  #
+                    # checks are performed in this order,    #
+                    # otherwise the 'boundaries' array will  #
```

Comment:
Not sure if this comment is needed as order does often matter in an if/elif block.

Reply:
Agreed, this does seem unnecessary. I've removed this comment in the latest commit.